pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1516,6 +1523,9 @@ func (c *Cluster) startMon(m *monConfig, schedule *controller.MonScheduleInfo) e
 		return err
 	}
 
+	// apply cephx secret resource version to the deployment to ensure it restarts after key is rotated
+	d.Spec.Template.Annotations[keyring.CephxKeyIdentifierAnnotation] = c.monKeySecretRef
```

Comment:
In upgraded clusters, if this annotation is always added, it will cause the mon restart even if we are not rotating. If we could avoid adding this annotation until the first rotation, it would be nice to avoid the unnecessary restart. But if it's not a simple change, don't worry about it.

Reply:
won't the cluster upgrade and addition of the new annotation happen at the same time. So it just will be one mon restart both for upgrading the mons and as well as adding the annotation.  We can discuss more about this in the huddle.

Reply:
A Rook upgrade only causes an update to the spec if something with the pod spec changed. During upgrades, this only happens occasionally. I don't see an easy way to know if the rotation has ever happened, so let's not worry about it.